counterparty-core/counterpartycore/lib/follow.py

```diff
@@ -182,9 +182,11 @@ async def receive_multipart(self, socket, topic_name):
         try:
             flags = 0 if topic_name == "sequence" else zmq.NOBLOCK
             topic, body, seq = await socket.recv_multipart(flags=flags)
-        except zmq.ZMQError:
-            logger.trace("No message available in topic `%s`", topic_name)
-            return
+        except zmq.ZMQError as e:
+            if e.errno == zmq.EAGAIN:
+                logger.trace("No message available in topic `%s`", topic_name)
+                return
+            raise e
```

Comment:
if the error gets raised, will the client restart?

Reply:
If there is an error other than zmq.EAGAIN, we must first identify it to react accordingly.